### PR TITLE
docs(build): add text build steps for Linux, Mac and Windows

### PR DESCRIPTION
**Pull request by AbhinavMir**
Issue: #1963

Signed-off-by: Abhinav Srivastava (abhinavmir#1898 on Discord)

### CODE REVIEW

**Review by petermetz — CHANGES_REQUESTED** [review]
@AbhinavMir 

1. apt is only there on Debian and it's downstream distros so I would somehow make it clear and then in separate sections you could detail the steps for Win and Mac so that people don't get confused.
2. Instead of npm, please use yarn where possible
3. Do not install the `nodejs` package via apt (not even on the Debian distros). We strongly encourage/prefer the use of the node version manager tool (nvm)
4. Let's push people towards JDK 11 instead of JDK 8

**Comment by AbhinavMir** [comment]
have addressed 2,3,4 in latest commit. I think for 1, we can create a new issue to create separate build instructions on Windows and Mac.

File: BUILD.md [hunk 1]
```diff
@@ -1,91 +1,91 @@
-# Hyperledger Cactus Build Instructions
+### Cactus BUILD instruction
```
Comment (jagpreetsinghsasan) [inline BUILD.md:1]: I believe saying Hyperledger Cactus (instead of just Cactus) here is better.

File: BUILD.md [hunk 2]
```diff
@@ -1,91 +1,91 @@
...
+curl -fsSL https://download.docker.com/linux/debian/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/debian \ $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+  
+### Install the Docker Engine
+sudo apt-get update
+sudo apt-get install docker-ce docker-ce-cli containerd.io docker-compose-plugin
+
+### Check that the package has been correctly installed
+sudo docker run hello-world
```
Comment (jagpreetsinghsasan) [inline BUILD.md:47]: After this line, reference can also be given to use docker without sudo, as mentioned here, https://docs.docker.com/engine/install/linux-postinstall/

File: BUILD.md [hunk 3]
```diff
@@ -1,91 +1,91 @@
...
 
-We put a lot of thought and effort into making sure that fast developer iterations can be
-achieved (please file a bug if you feel otherwise) while working **on** the framework.
+#### Installing VS Code 
 
-If you find yourself waiting too much for builds to finish, most of the time
-that can be helped by using the `npm run watch` script which can automatically
-recompile packages as you modify them (and only the packages that you have
-modified, not everything).
+You can download and install VS Code from [here](https://code.visualstudio.com/download). It is suggested to install the git-graph and Docker plugins.
```
Comment (jagpreetsinghsasan) [inline BUILD.md:77]: Here, we can also mention the `npm-scripts` plugin as well (https://marketplace.visualstudio.com/items?itemName=traBpUkciP.vscode-npm-scripts)

File: BUILD.md [hunk 4]
```diff
@@ -1,91 +1,91 @@
...
+You can download and install VS Code from [here](https://code.visualstudio.com/download). It is suggested to install the git-graph and Docker plugins. 
 
-It also supports re-running the OpenAPI generator when you update any
-`openapi.json` spec files that we use to describe our endpoints.
+#### Clone and compile the Cactus repository 🌵
 
-The `npm run watch` script in action:
+```
+git clone https://github.com/hyperledger/cactus
+yarn add
```
Comment (jagpreetsinghsasan) [inline BUILD.md:83]: We can specify the `npm run install-yarn` prior to`yarn add` (yarn-add isn't required)

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]

**Pull request closed without merging**